### DISCUSSION TITLE
using local directory as bucket?

### DISCUSSION THREAD

**Discussion by goyalyashpal** [ScoopInstaller/Scoop#5867] [Help and Questions] [answered]
### 1. i was thinking if a local bucket could be used as a bucket too?

as otherwise, `scoop` is kinda not linking back the installs with their locally available manifests 😅 for `scoop info` or `scoop search`, despite knowing manifest location in `scoop list`

```console
$ scoop update neovim-dev
neovim-dev: 0.10.0-dev-2801 (latest version)
Latest versions for all apps are installed! For more information try 'scoop status'

$ scoop info !$
scoop info neovim-dev
Could not find manifest for 'neovim-dev' in local buckets.

$ scoop search !$
scoop search neovim-dev
WARN  No matches found.
```

while in reality, `neovim-dev` very much exists

```console
$ scoop list !$
scoop list neovim-dev
Installed apps matching 'neovim-dev':

Name       Version         Source                                Updated             Info
----       -------         ------                                -------             ----
neovim-dev 0.10.0-dev-2801 D:\5_setups\manifests\neovim-dev.json 2024-04-05 13:18:07

$ scoop info $manif/neovim-dev

Name        : neovim-dev
Description : Vim-fork focused on extensibility and usability
Version     : 0.10.0-dev-2801
...

```

https://github.com/ScoopInstaller/Scoop/wiki/Buckets

**Comment by niheaven** [comment, marked as answer]
You should place the local bucket directory inside the Scoop's `buckets` directory, and your manifests should be stored under that directory or in a subdirectory named `bucket`. Refer to https://github.com/ScoopInstaller/BucketTemplate for the bucket structure.

**Reply by goyalyashpal** [reply]
hi! that looks nice & [`scoop bucket list`][bucket list] works 😃 .

[bucket list]: https://github.com/ScoopInstaller/Scoop/wiki/Buckets#after-the-bucket-is-created

but 2 questions:
1. any suggestion for the directory name for it? 😅 

2. `scoop list neovim-dev` still shows the "source" to be old dir which is understandable. but how to change it?

* i tried doing `scoop install neovim-dev`, it showed `... already installed. Use 'scoop update ...' ...`
* on `scoop update neovim-dev` or even `--force` it says: `WARN  error: Could not find a part of the path '<path-to-old-location>'.` `ERROR No manifest available for 'neovim-dev'.`
* i even tried specifying bucket name: `scoop update manifests/neovim-dev --force` and `scoop list manifests/neovim-dev` still no go
* `scoop reset manifests/neovim-dev` didn't fix it either

**Reply by niheaven** [reply]
AHA,yes,you must uninstall and install your old app,since scoop will remember your old manifest and file location in `install.json`. So reinstalling doesn't work, just uninstall and install again.

**Reply by goyalyashpal** [reply]
ohw, ohkayh. 

got it. thanks a lot 😃
